ocaml/parsing/pprintast.ml

```diff
@@ -1461,16 +1480,18 @@ and binding ctxt f {pvb_pat=p; pvb_expr=x; _} =
 and bindings ctxt f (rf,l) =
   let binding kwd rf f x =
     let x, is_local =
-      match x.pvb_expr.pexp_desc with
-      | Pexp_apply
-        ({ pexp_desc = Pexp_extension({txt = "extension.local"}, PStr []) },
-         [Nolabel, sbody]) ->
-         let sattrs, _ = check_local_attr sbody.pexp_attributes in
-         let sbody = {sbody with pexp_attributes = sattrs} in
-         let pattrs, _ = check_local_attr x.pvb_pat.ppat_attributes in
-         let pat = {x.pvb_pat with ppat_attributes = pattrs} in
-         {x with pvb_pat = pat; pvb_expr = sbody}, "local_ "
-      | _ -> x, ""
+      match Jane_syntax.Expression.of_ast x.pvb_expr with
+      | Some (Jexp_local (Lexp_local expr), attrs) ->
+          let pat, pat_attrs = match Jane_syntax.Pattern.of_ast x.pvb_pat with
+            | Some (Jpat_local (Lpat_local lpat), pat_attrs) -> lpat, pat_attrs
+            | Some _ | None -> x.pvb_pat, x.pvb_attributes
+          in
+          let x = {x with pvb_pat = pat;
+                          pvb_attributes = pat_attrs;
+                          pvb_expr = { expr with pexp_attributes = attrs } }
```

Comment:
I believe this is also a bug: we shouldn't put `attrs` (the outer attributes on the overall local node) onto the inner expression. The bug exhibits thusly:

```ocaml
$ nroberts-out/bin/ocaml -dsource -w -A
OCaml version 4.14.1+jst
Enter #help;; for help.

# let () =
    let x = (local_ 3)[@outer] in
    ()
  ;;

let () = let local_ x = ((3)[@outer ]) in ();;
```
That is, `-dsource` shows how the `outer` attribute is being transplanted on `inner`. Likely, we should avoid doing this rewrite in `bindings` if `attrs` is non-empty, as then there's no way to rewrite into `let local_ x = ...`.

Reply:
Yeah, that sounds right -- fixed